app/components/formatted_time.js

```diff
@@ -3,35 +3,32 @@
 
 import React from 'react';
 import PropTypes from 'prop-types';
+import {intlShape} from 'react-intl';
 import {Text} from 'react-native';
 
 export default class FormattedTime extends React.PureComponent {
     static propTypes = {
         value: PropTypes.any.isRequired,
+        timeZone: PropTypes.string,
         children: PropTypes.func,
         hour12: PropTypes.bool,
     };
 
+    static contextTypes = {
+        intl: intlShape.isRequired,
+    };
+
     render() {
         const {
             value,
             children,
-            hour12,
+            timeZone,
+            ...props
         } = this.props;
+        const {intl} = this.context;
 
-        const date = new Date(value);
-        const militaryTime = !hour12;
-
-        const hour = militaryTime ? date.getHours() : (date.getHours() % 12 || 12);
-        let minute = date.getMinutes();
-        minute = minute >= 10 ? minute : ('0' + minute);
-        let time = '';
-
-        if (!militaryTime) {
-            time = (date.getHours() >= 12 ? ' PM' : ' AM');
-        }
-
-        const formattedTime = hour + ':' + minute + time;
+        const timezoneProps = timeZone ? {timeZone} : {};
+        const formattedTime = intl.formatDate(value, {...props, ...timezoneProps, hour: 'numeric', minute: 'numeric'});
```

Comment:
Regarding the `...props`, are we actually passing extra props into this component that we'd want to pass on? I can't find anywhere that we're doing that.

Reply:
I believe the only thing being passed in `...props` is `hour12`.

I just left it as-is from previous implementation before the refactor to simple math

Reply:
@hmhealey do you want this to be an explicit prop pass?

Reply:
As long as this doesn't cause any problems when an unrecognized prop is passed, I think this is fine since it won't cause unexpected issues later on. We might want to confirm that it doesn't though.